hw3/02-charts.js

```diff
@@ -1,44 +1,66 @@
 const backgroundColors = [
-  'rgba(54, 162, 235, 0.8)',
-  'rgba(255, 206, 86, 0.8)',
-  'rgba(255, 99, 132, 0.8)',
-  'rgba(75, 192, 192, 0.8)',
-  'rgba(153, 102, 255, 0.8)',
-  'rgba(255, 159, 64, 0.8)',
-  'rgba(199, 199, 199, 0.8)',
-  'rgba(83, 102, 255, 0.8)',
-  'rgba(40, 159, 64, 0.8)',
-  'rgba(210, 199, 199, 0.8)',
-  'rgba(78, 52, 199, 0.8)',
+  "rgba(54, 162, 235, 0.8)",
+  "rgba(255, 206, 86, 0.8)",
+  "rgba(255, 99, 132, 0.8)",
+  "rgba(75, 192, 192, 0.8)",
+  "rgba(153, 102, 255, 0.8)",
+  "rgba(255, 159, 64, 0.8)",
+  "rgba(199, 199, 199, 0.8)",
+  "rgba(83, 102, 255, 0.8)",
+  "rgba(40, 159, 64, 0.8)",
+  "rgba(210, 199, 199, 0.8)",
+  "rgba(78, 52, 199, 0.8)",
 ];
 
 const borderColors = [
-  'rgba(54, 162, 235, 1)',
-  'rgba(255, 206, 86, 1)',
-  'rgba(255, 99, 132, 1)',
-  'rgba(75, 192, 192, 1)',
-  'rgba(153, 102, 255, 1)',
-  'rgba(255, 159, 64, 1)',
-  'rgba(159, 159, 159, 1)',
-  'rgba(83, 102, 255, 1)',
-  'rgba(40, 159, 64, 1)',
-  'rgba(210, 199, 199, 1)',
-  'rgba(78, 52, 199, 1)',
+  "rgba(54, 162, 235, 1)",
+  "rgba(255, 206, 86, 1)",
+  "rgba(255, 99, 132, 1)",
+  "rgba(75, 192, 192, 1)",
+  "rgba(153, 102, 255, 1)",
+  "rgba(255, 159, 64, 1)",
+  "rgba(159, 159, 159, 1)",
+  "rgba(83, 102, 255, 1)",
+  "rgba(40, 159, 64, 1)",
+  "rgba(210, 199, 199, 1)",
+  "rgba(78, 52, 199, 1)",
 ];
 
 // url for the Thrones API
-const url = 'https://thronesapi.com/api/v2/Characters';
+const url = "https://thronesapi.com/api/v2/Characters";
+
+//to retrieve all the characters in that API
+function updateChart() {
+  async function logJSONData() {
+    const response = await fetch(url);
+    //wait until the request has been completed
+    const jsonData = await response.json();
+    console.log(jsonData);
+    return jsonData;
+  }
+
+  logJSONData().then((jsonData) => {
+    const family = jsonData.family.map(function (index) {
+      return index.fullName;
+    });
+    console.log(family);
+
+    //update labels
+    renderChart.Chart.data.labels = family;
+    renderChart.update();
+  });
+}
 
 const renderChart = () => {
-  const donutChart = document.querySelector('.donut-chart');
+  const donutChart = document.querySelector(".donut-chart");
 
   new Chart(donutChart, {
-    type: 'doughnut',
+    type: "doughnut",
     data: {
-      labels: ['label', 'label', 'label', 'label'],
+      labels: ["label", "label", "label", "label"],
```

Comment:
The general approach I used here was to create an object with the API data whose keys were house names and whose values were integer counts indicating occurrences of each house in the data set. Then you can use something like `Object.keys(data)` for the `labels` field here and `Object.keys(values)` for the `data` field a few lines below.